index.js

```diff
@@ -826,6 +888,18 @@
     return function(x) { return f(chain(x))(x); };
   }
 
+  //  Function$prototype$alt :: (x -> a) ~> (x -> a) -> x -> a
+  function Function$prototype$alt(f) {
+    var alt_ = this;
+    return function(x) { return alt(alt_(x), f(x)); };
+  }
+
+  //  Function$prototype$concat :: (x -> a) ~> (x -> a) -> x -> a
+  function Function$prototype$concat(f) {
+    var semigroup = this;
+    return function(x) { return concat(semigroup(x), f(x)); };
+  }
+
```

Comment:
No, only `this.concat(that)` - in both cases, we have associativity, but not necessarily commutativity. e.g.:

```haskell
Just [2] <> Just [3] == Just [2, 3]
(const $ Just [2]) <> (const $ Just [3]) == const $ Just [2, 3]
```

Just as we can't change ordering of `Array` concat within `Maybe` concat, we can't change ordering of `Array` concat within `Maybe` concat within `Function` concat!